File: src/completed/java/euler0005.java
/**
 * Created by Holden on 11/19/2016. - COMPLETED
 *
 * Problem 5 - Smallest Multiple
 *
 */

public class Prob0005 {
    public static void main(String[] args) {
        long startTime = System.nanoTime();
        int answer = smallestMultiple(20);
        long endTime = System.nanoTime();

        System.out.println(answer);

        long duration = (endTime - startTime);
        EulerTime.time(duration);
    }

    public static int smallestMultiple(int upperBound) {
        // Construct list of multiples
        OccurrenceSet<Integer> multiples = new OccurrenceSet<Integer>();
        for (int i = 2; i <= upperBound; i++) {
            multiples.unionAdd(new OccurrenceSet<Integer>(EulerUtil.primeFactor(i)));
        }

        int product = 1;
        for (Integer i: multiples) {
            product *= EulerUtil.pow(i, multiples.get(i));
        }

        return product;
    }
}

File: src/completed/python/euler0005.py
###
#
# Created by Holden on 12/17/2015
#
# SOLVED on 12/17/2015
#
# Problem 5 - Smallest Multiple
#
###

def smallest_multiple(products):
    n = 1
    primes = []
    
    for i in range(2, products + 1):
        for prime in primes:
            if i % prime == 0:
                i = int(i / prime)
            else:
                pass
        primes.append(i)
        n *= i
        
    return n